tests/sdk/test_client.py

```diff
@@ -0,0 +1,188 @@
+        return "env_var_token"
+    if username == "toto_file_username" and password == "toto_file_password":
+        return "toto_file_token"
+
+
+@pytest.mark.parametrize(
+    ["input", "expected"],
+    [
+        ("toto", "TOTO"),
+        ("client-org-1", "CLIENT_ORG_1"),
+        ("un nom très français", "UN_NOM_TRES_FRANCAIS"),
+    ],
+)
+def test_upper_slug(input, expected):
+    assert expected == _upper_slug(input)
+
+
+def test_default_client():
+    client = Client()
+    assert client.backend_mode == schemas.BackendType.LOCAL_SUBPROCESS
+
+
+@pytest.mark.parametrize(
+    ["mode", "configuration_name", "url", "token", "insecure", "retry_timeout"],
+    [
+        (schemas.BackendType.LOCAL_SUBPROCESS, "foo", None, None, True, 5),
+        (schemas.BackendType.LOCAL_DOCKER, "foobar", None, None, True, None),
+        (schemas.BackendType.REMOTE, "bar", "example.com", "bloop", False, 15),
+        (schemas.BackendType.REMOTE, "hybrid", "example.com", "foo", True, 500),
+    ],
+)
+def test_client_configured_in_code(mode, configuration_name, url, token, insecure, retry_timeout):
+    client = Client(
+        backend_type=mode,
+        configuration_name=configuration_name,
+        url=url,
+        token=token,
+        insecure=insecure,
+        retry_timeout=retry_timeout,
+    )
+    assert client.backend_mode == mode
+    assert client._insecure == insecure
+    if retry_timeout is not None:
+        assert client._retry_timeout == retry_timeout
+    else:
+        assert client._retry_timeout == 300
+    if token is not None:
+        assert client._token == token
+    else:
+        assert client._token is None
+    if url is not None:
+        assert client._url == url
+    else:
+        assert client._url is None
+
+
+def test_client_with_password(mocker):
+    mocker.patch("substra.sdk.Client.login", side_effect=stub_login)
+    client = Client(
+        backend_type="remote",
+        url="example.com",
+        token=None,
+        username="org-1",
+        password="password1",
+    )
+    assert client._token == "token1"
+
+
+def test_client_token_supercedes_password(mocker):
+    mocker.patch("substra.sdk.Client.login", side_effect=stub_login)
+    client = Client(
+        backend_type="remote",
+        url="example.com",
+        token="token0",
+        username="org-1",
+        password="password1",
+    )
+    assert client._token == "token0"
+
+
+def test_client_configuration_from_env_var(mocker, monkeypatch):
+    mocker.patch("substra.sdk.Client.login", side_effect=stub_login)
+    monkeypatch.setenv("SUBSTRA_TOTO_BACKEND_TYPE", "remote")
+    monkeypatch.setenv("SUBSTRA_TOTO_URL", "toto-org.com")
+    monkeypatch.setenv("SUBSTRA_TOTO_USERNAME", "env_var_username")
+    monkeypatch.setenv("SUBSTRA_TOTO_PASSWORD", "env_var_password")
+    monkeypatch.setenv("SUBSTRA_TOTO_RETRY_TIMEOUT", "42")
+    monkeypatch.setenv("SUBSTRA_TOTO_INSECURE", "true")
+    client = Client(configuration_name="toto")
```

Comment:
What does the configuration_name in that case ?

Reply:
missing a verb in this question I think 😅

Reply:
Woooops ^^" What does the configuration_name does in that case ?

Reply:
I mean what the expected behaviour of passing a config name without a config file?

Reply:
It is used to build the env var names, such as `SUBSTRA_TOTO_PASSWORD`. 
I was wondering if it was necessary to add an how-to guide for this feature, but given the many questions you have, I guess mine is answered 😅

Reply:
Aaaaah ok ok I understand. Documentation might be welcome yes ahah :)